src/main/java/duke/TasksList.java

```diff
@@ -0,0 +1,166 @@
+        try {
+            String description = rawInput.substring(LENGTH_OF_TODO);
+            tasks.add(new ToDo(description));
+            System.out.println("\tGot it. I've added this task:");
+            System.out.println("\t\t" + this.getTaskString(this.getSize() - 1));
+            System.out.println("\tNow you have " + this.getSize() + " tasks in the list.");
+        } catch (StringIndexOutOfBoundsException e) {
+            System.out.println("\tOOPS!!! The description of a todo cannot be empty.");
+        }
+    }
+
+    public void addEvent(String rawInput) {
+        try{
+            int idx = rawInput.indexOf("/at");
+            String description = rawInput.substring(LENGTH_OF_EVENT, idx-1);
+            String at = rawInput.substring(idx+LENGTH_OF_AT);
+            tasks.add(new Event(description, at));
+            System.out.println("\tGot it. I've added this task:");
+            System.out.println("\t\t" + this.getTaskString(this.getSize() - 1));
+            System.out.println("\tNow you have " + this.getSize() + " tasks in the list.");
+        } catch (StringIndexOutOfBoundsException e) {
+            System.out.println("\tWrong deadline format! Please follow `deadline <description> /at <at>`.");
+        }
+    }
+
+    public void addDeadline(String rawInput) {
+        try {
+            int idx = rawInput.indexOf("/by");
+            String description = rawInput.substring(LENGTH_OF_DEADLINE, idx-1);
+            String by = rawInput.substring(idx+LENGTH_OF_BY);
+            tasks.add(new Deadline(description, by));
+            System.out.println("\tGot it. I've added this task:");
+            System.out.println("\t\t" + this.getTaskString(this.getSize() - 1));
+            System.out.println("\tNow you have " + this.getSize() + " tasks in the list.");
+        } catch (StringIndexOutOfBoundsException e) {
+            System.out.println("\tWrong deadline format! Please follow `deadline <description> /by <by>`.");
+        }
+    }
+
+    public void markTaskAsDone(String rawInput) {
+        try {
+            int taskIndex = Integer.parseInt(rawInput.substring(5)) - 1;
+            Task task = tasks.get(taskIndex);
+            System.out.println("\tNice! I've marked this task as done:");
+            task.markAsDone();
+            System.out.println("\t\t[" + task.getStatusIcon() + "] " + task.getDescription());
+        } catch (NumberFormatException e) {
+            System.out.println("\tWrong format, please retype with format `done <taskIndex>`.");
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println("\t Task index is out of range.");
+        }
+    }
+
+    public void deleteTask(String rawInput) {
+        int taskIndex = Integer.parseInt(rawInput.substring(7)) - 1;
+        try {
+            Task task = tasks.get(taskIndex);
+            System.out.println("\tNoted. I have removed this task:");
+            System.out.println("\t\t" + task.toString());
```

Comment:
Can consider putting the printing message behaviour into another function for better SLAP.